src/main/java/uk/gov/companieshouse/overseasentitiesapi/service/ManagingOfficerChangeService.java

```diff
@@ -0,0 +1,291 @@
+                CompanyOfficerApi::getPrincipalOfficeAddress,
+                TypeConverter::addressDtoToAddress,
+                ComparisonHelper::equals,
+                CorporateManagingOfficer::setRegisteredOffice
+        );
+
+        hasChange |= changeManager.compareAndBuildRightChange(
+                managingOfficerCorporateDto.getContactFullName(),
+                ManagingOfficerDataApi::getContactNameFull,
+                CorporateManagingOfficer::setContactName
+        );
+
+        hasChange |= changeManager.compareAndBuildRightChange(
+                managingOfficerCorporateDto.getContactEmail(),
+                ManagingOfficerDataApi::getContactEmailAddress,
+                CorporateManagingOfficer::setEmail
+        );
+
+        officer.setCompanyIdentification(new CompanyIdentification());
+
+        IdentificationApi publicIdentification = publicPrivateMoPair.getLeft() == null ? null
+                : publicPrivateMoPair.getLeft().getIdentification();
+
+        ChangeManager<CompanyIdentification, CompanyOfficerApi, IdentificationApi> companyIdentificationChangeManager
+                = new ChangeManager<>(officer.getCompanyIdentification(),
+                new ImmutablePair<>(publicPrivateMoPair.getLeft(), publicIdentification));
+
+        hasChange |= companyIdentificationChangeManager.compareAndBuildRightChange(
+                managingOfficerCorporateDto.getLegalForm(),
+                IdentificationApi::getLegalForm,
+                CompanyIdentification::setLegalForm);
+
+        hasChange |= companyIdentificationChangeManager.compareAndBuildRightChange(
+                managingOfficerCorporateDto.getLawGoverned(),
+                IdentificationApi::getLegalAuthority,
+                CompanyIdentification::setGoverningLaw);
+
+        hasChange |= companyIdentificationChangeManager.compareAndBuildRightChange(
+                managingOfficerCorporateDto.getPublicRegisterName(),
+                IdentificationApi::getPlaceRegistered,
+                CompanyIdentification::setRegisterLocation);
+
+        hasChange |= companyIdentificationChangeManager.compareAndBuildRightChange(
+                managingOfficerCorporateDto.getRegistrationNumber(),
+                IdentificationApi::getRegistrationNumber,
+                CompanyIdentification::setRegistrationNumber);
+
+        return hasChange ? managingOfficerChange : null;
+    }
+
+    //todo: add comment
+    private <P extends Officer> boolean setCommonAttributes(
+            ChangeManager<P, CompanyOfficerApi, ManagingOfficerDataApi> changeManager,
+            AddressDto serviceAddress,
+            LocalDate actionDate) {
+        var hasChange = changeManager.compareAndBuildRightChange(
+                serviceAddress,
+                ManagingOfficerDataApi::getPrincipalAddress,
+                TypeConverter::addressDtoToAddress,
+                ComparisonHelper::equals,
+                Officer::setServiceAddress
+        );
+
```

Comment:
We'll communicate the resigned on in cessations. So not needed here. It's an optional field so the comparison is a bit tricky as we can't pass a "null" action date to CHIPS.